dev/tasks/push-cloud-controller-manager

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -x
+
+source $(git rev-parse --show-toplevel)/dev/tasks/common
+
+# IMAGE_REPO is used to upload images
+if [[ -z "${IMAGE_REPO:-}" ]]; then
+  echo "IMAGE_REPO must be set"
+  exit 1
+fi
+echo "IMAGE_REPO=${IMAGE_REPO}"
+export KO_DOCKER_REPO="${IMAGE_REPO}"
+
+# We default IMAGE_TAG to a unique value that combines the sha and the (real) build date
+if [[ -z "${IMAGE_TAG:-}" ]]; then
+  IMAGE_TAG=$(git rev-parse --short HEAD)-$(date +%Y%m%dT%H%M%S)
+fi
+echo "IMAGE_TAG=${IMAGE_TAG}"
+
+
+KO_CONFIG_PATH=$(mktemp ${WORKSPACE}/cloud-controller-manager.XXXX.ko.yaml)
+export KO_CONFIG_PATH
+
+cat > ${KO_CONFIG_PATH} <<EOF
+builds:
+- id: cloud-controller-manager
+  dir: .
+  main: cmd/cloud-controller-manager
+  ldflags:
+  - -X k8s.io/component-base/version/verflag.programName=cloud-controller-manager
+
+  - -X k8s.io/component-base/version.buildDate=${BUILD_DATE}
+  - -X k8s.io/component-base/version.gitCommit=${KUBE_GIT_COMMIT}
+  - -X k8s.io/component-base/version.gitMajor=${KUBE_GIT_MAJOR}
+  - -X k8s.io/component-base/version.gitMinor=${KUBE_GIT_MINOR}
+  - -X k8s.io/component-base/version.gitTreeState=${KUBE_GIT_TREE_STATE}
+  - -X k8s.io/component-base/version.gitVersion=${KUBE_GIT_VERSION}
+
+  - -X k8s.io/client-go/pkg/version.buildDate=${BUILD_DATE}
+  - -X k8s.io/client-go/pkg/version.gitCommit=${KUBE_GIT_COMMIT}
+  - -X k8s.io/client-go/pkg/version.gitMajor=${KUBE_GIT_MAJOR}
+  - -X k8s.io/client-go/pkg/version.gitMinor=${KUBE_GIT_MINOR}
+  - -X k8s.io/client-go/pkg/version.gitTreeState=${KUBE_GIT_TREE_STATE}
+  - -X k8s.io/client-go/pkg/version.gitVersion=${KUBE_GIT_VERSION}
+EOF
+
+go run github.com/google/ko@v0.14.1 build --platform=linux/amd64,linux/arm64 --tags=${IMAGE_TAG} --base-import-paths --push=true ./cmd/cloud-controller-manager/
```

Comment:
we already have cloudbuild, can not we reuse it?

Reply:
We will :-)  ko replaces go build / docker build / sbom / signing etc, not cloudbuild.

Reply:
Here's how it can work: https://github.com/kubernetes/cloud-provider-gcp/pull/600